src/main/java/com/mycompany/shop/basket/BasketController.java

```diff
@@ -33,21 +39,75 @@
     static class CreateBasketRequest {
         Long id;
 
+        Long userId;
+
         public Long getId() {
             return id;
         }
 
         public void setId(Long id) {
             this.id = id;
         }
+
+        public Long getUserId() {
+            return userId;
+        }
+
+        public void setUserId(Long userId) {
+            this.userId = userId;
+        }
     }
 
     @GetMapping
+    public ResponseEntity<Basket> getUserBasket(@RequestParam("userId") Long userId) {
+        Basket basket;
+        if (userId != null) {
+            if (basketRepository.findBasketByUserId(userId).isPresent()) {
+                basket = basketRepository.findBasketByUserId(userId).get();
+                sessionBean.setBasket(basket);
+            } else {
+                sessionBean.setBasket(null);
+            }
```

Comment:
that logic should be done on login step. also here I can replace userId by any userId, so I can get basket of any user in application

Reply:
+